### PR TITLE
fix(gnoland): Update error type for invalid deposit

### PR DESCRIPTION
**Pull request by linhpn99**
Should use a more appropriate error type

<details><summary>Contributors' checklist...</summary>

- [ ] Added new tests, or not needed, or not feasible
- [ ] Provided an example (e.g. screenshot) to aid review or the PR is self-explanatory
- [ ] Updated the official documentation or not needed
- [ ] No breaking changes were made, or a `BREAKING CHANGE: xxx` message was included in the description
- [ ] Added references to related issues and PRs
- [ ] Provided any useful hints for running manual tests
- [ ] Added new benchmarks to [generated graphs](https://gnoland.github.io/benchmarks), if any. More info [here](https://github.com/gnolang/gno/blob/master/.benchmarks/README.md).
</details>

### CODE REVIEW

File: gno.land/pkg/sdk/vm/msgs.go [hunk 1]
```diff
@@ -57,7 +57,7 @@ func (msg MsgAddPackage) ValidateBasic() error {
 		return ErrInvalidPkgPath("missing package path")
 	}
 	if !msg.Deposit.IsValid() {
-		return std.ErrTxDecode("invalid deposit")
+		return std.ErrInvalidCoins("invalid deposit")
```
Comment (thehowl) [inline gno.land/pkg/sdk/vm/msgs.go:60]: most other places using ErrInvalidCoins do it with `std.ErrInvalidCoins(amt.String())` -- so please do the same

Comment (linhpn99) [reply gno.land/pkg/sdk/vm/msgs.go]: ff3ad2350f15e897488baa4cbf2ea261695a2899

File: tm2/pkg/sdk/bank/msgs.go
Comment (linhpn99) [inline tm2/pkg/sdk/bank/msgs.go:1]: Delete to be consistent with the existing error handling in this file

File: tm2/pkg/sdk/bank/msgs.go [hunk 2]
```diff
@@ -39,7 +39,7 @@ func (msg MsgSend) ValidateBasic() error {
 		return std.ErrInvalidAddress("missing recipient address")
 	}
 	if !msg.Amount.IsValid() {
-		return std.ErrInvalidCoins("send amount is invalid: " + msg.Amount.String())
```
Comment (linhpn99) [inline tm2/pkg/sdk/bank/msgs.go:42]: Delete to be consistent with the existing error handling in this file

File: tm2/pkg/sdk/bank/msgs.go [hunk 3]
```diff
@@ -39,7 +39,7 @@ func (msg MsgSend) ValidateBasic() error {
...
+		return std.ErrInvalidCoins(msg.Amount.String())
```
Comment (linhpn99) [inline tm2/pkg/sdk/bank/msgs.go:42]: Delete to be consistent with the existing error handling in this file

**Review by thehowl — APPROVED** [review]

**Pull request merged**